test_conformance/extensions/cl_khr_external_semaphore/test_external_semaphore_handle_types.cpp

```diff
@@ -0,0 +1,178 @@
+        static_cast<cl_semaphore_properties_khr>(CL_SEMAPHORE_TYPE_KHR),
+        static_cast<cl_semaphore_properties_khr>(CL_SEMAPHORE_TYPE_BINARY_KHR),
+        static_cast<cl_semaphore_properties_khr>(
+            CL_SEMAPHORE_EXPORT_HANDLE_TYPES_KHR),
+        static_cast<cl_semaphore_properties_khr>(handle_type),
+        static_cast<cl_semaphore_properties_khr>(
+            CL_SEMAPHORE_EXPORT_HANDLE_TYPES_LIST_END_KHR),
+        0
+    };
+    cl_semaphore_khr sema_1 =
+        clCreateSemaphoreWithPropertiesKHR(context, sema_1_props, &err);
+    test_error(err, "Could not create semaphore");
+
+    // Signal semaphore
+    clEventWrapper signal_event;
+    err = clEnqueueSignalSemaphoresKHR(queue, 1, &sema_1, nullptr, 0, nullptr,
+                                       &signal_event);
+    test_error(err, "Could not signal semaphore");
+
+    // Extract sync fd
+    int handle = -1;
+    size_t handle_size;
+    err = clGetSemaphoreHandleForTypeKHR(sema_1, device, handle_type,
+                                         sizeof(handle), &handle, &handle_size);
+    test_error(err, "Could not extract semaphore handle");
+    test_assert_error(sizeof(handle) == handle_size, "Invalid handle size");
+    test_assert_error(handle >= 0, "Invalid handle");
+
+    // Create semaphore from sync fd
+    cl_semaphore_properties_khr sema_2_props[] = {
+        static_cast<cl_semaphore_properties_khr>(CL_SEMAPHORE_TYPE_KHR),
+        static_cast<cl_semaphore_properties_khr>(CL_SEMAPHORE_TYPE_BINARY_KHR),
+        handle_type, static_cast<cl_semaphore_properties_khr>(handle), 0
+    };
+
+    cl_semaphore_khr sema_2 =
+        clCreateSemaphoreWithPropertiesKHR(context, sema_2_props, &err);
+    test_error(err, "Could not create semaphore");
+
+    // Wait semaphore
+    clEventWrapper wait_event;
+    err = clEnqueueWaitSemaphoresKHR(queue, 1, &sema_2, nullptr, 0, nullptr,
+                                     &wait_event);
+    test_error(err, "Could not wait semaphore");
+
+    // Finish
+    err = clFinish(queue);
+    test_error(err, "Could not finish queue");
+
+    // Check all events are completed
+    test_assert_event_complete(signal_event);
+    test_assert_event_complete(wait_event);
+
+    // Release semaphore
+    err = clReleaseSemaphoreKHR(sema_1);
+    test_error(err, "Could not release semaphore");
+
+    err = clReleaseSemaphoreKHR(sema_2);
+    test_error(err, "Could not release semaphore");
+    return TEST_PASS;
+}
+
+// Test it is possible to export a semaphore and import the same to a new
+// semaphore
+REGISTER_TEST_VERSION(external_semaphores_import_export_fd, Version(1, 2))
+{
+    REQUIRE_EXTENSION("cl_khr_external_semaphore");
+
+    std::vector<cl_semaphore_properties_khr> handle_types = {
+        CL_SEMAPHORE_HANDLE_SYNC_FD_KHR, CL_SEMAPHORE_HANDLE_OPAQUE_FD_KHR,
+#ifdef _WIN32
+        CL_SEMAPHORE_HANDLE_D3D12_FENCE_KHR,
+        CL_SEMAPHORE_HANDLE_OPAQUE_WIN32_KHR
+#endif
```

Comment:
I don't think this will work out-of-the-box for these win32 handle types because they are NTHANDLE (void*) handle types, which are different than the fd (int) handle types expected by `doTest`.

Does this work for the OPAQUE_FD handle type, though?

If it does, let's get a PR merged with it first, then we can look to expand to the win32 handle types, if it makes sense.